src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncHelpers.cs

```diff
@@ -0,0 +1,175 @@
+            }
+
+            return awaiter.GetResult();
+        }
+
+        // Marked intrinsic since JIT recognises the helper by name when doing optimizations.
+        [Intrinsic]
+        [BypassReadyToRun]
+        [MethodImpl(MethodImplOptions.Async)]
+        public static void Await(ValueTask task)
+        {
+            ValueTaskAwaiter awaiter = task.GetAwaiter();
+            if (!awaiter.IsCompleted)
+            {
+                UnsafeAwaitAwaiter(awaiter);
+            }
+
+            awaiter.GetResult();
+        }
+
+        // Marked intrinsic since JIT recognises the helper by name when doing optimizations.
+        [Intrinsic]
+        [BypassReadyToRun]
+        [MethodImpl(MethodImplOptions.Async)]
+        public static void Await(ConfiguredTaskAwaitable configuredAwaitable)
+        {
+            ConfiguredTaskAwaitable.ConfiguredTaskAwaiter awaiter = configuredAwaitable.GetAwaiter();
+            if (!awaiter.IsCompleted)
+            {
+                UnsafeAwaitAwaiter(awaiter);
+            }
+
+            awaiter.GetResult();
+        }
+
+        // Marked intrinsic since JIT recognises the helper by name when doing optimizations.
+        [Intrinsic]
+        [BypassReadyToRun]
+        [MethodImpl(MethodImplOptions.Async)]
+        public static void Await(ConfiguredValueTaskAwaitable configuredAwaitable)
+        {
+            ConfiguredValueTaskAwaitable.ConfiguredValueTaskAwaiter awaiter = configuredAwaitable.GetAwaiter();
+            if (!awaiter.IsCompleted)
+            {
+                UnsafeAwaitAwaiter(awaiter);
+            }
+
+            awaiter.GetResult();
+        }
+
+        // Marked intrinsic since JIT recognises the helper by name when doing optimizations.
+        [Intrinsic]
+        [BypassReadyToRun]
+        [MethodImpl(MethodImplOptions.Async)]
+        public static T Await<T>(ConfiguredTaskAwaitable<T> configuredAwaitable)
+        {
+            ConfiguredTaskAwaitable<T>.ConfiguredTaskAwaiter awaiter = configuredAwaitable.GetAwaiter();
+            if (!awaiter.IsCompleted)
+            {
+                UnsafeAwaitAwaiter(awaiter);
+            }
+
+            return awaiter.GetResult();
+        }
+
+        // Marked intrinsic since JIT recognises the helper by name when doing optimizations.
+        [Intrinsic]
+        [BypassReadyToRun]
+        [MethodImpl(MethodImplOptions.Async)]
+        public static T Await<T>(ConfiguredValueTaskAwaitable<T> configuredAwaitable)
+        {
+            ConfiguredValueTaskAwaitable<T>.ConfiguredValueTaskAwaiter awaiter = configuredAwaitable.GetAwaiter();
+            if (!awaiter.IsCompleted)
+            {
+                UnsafeAwaitAwaiter(awaiter);
+            }
+
+            return awaiter.GetResult();
+        }
+#else
+        public static void UnsafeAwaitAwaiter<TAwaiter>(TAwaiter awaiter) where TAwaiter : ICriticalNotifyCompletion { throw new NotImplementedException(); }
```

Comment:
I am using this (as opposed to compat suppression) as suppressions do not seem to work with Mono.

If the enablement of the feature is tied to the presence of the API, then omitting helpers and suppressing compat failures could be a better option.

Reply:
The build and packaging system requires reference assemblies to be identical between CoreCLR vs. Mono. If you suppressed compat failures, you would still get the AsyncHelpers type in reference assembly. The only difference would be MissingMethodException vs. NotImplementedException at runtime if these APIs get called.

Reply:
Previously we had suppression for NativeAOT and helpers only for CoreCLR and clr partition could build. But not mono. The compat pass complains about diffs with ref and recommends to generate a suppression file, but generated file somehow does not have any effect and mono still fails to build. 
I suspected the location of the file may be incorrect, but could not find another suppression file under mono. 

There are some unimplemented public APIs, like under GC, but they appear to throw NotSupported or similar.

Reply:
I just want mono to build at this point (and pass existing tests, of course). 

In runtimelabs it is not a problem since  we build/test locally, but in main repo we need no regressions in mono.